### PR TITLE
support falling back to a bridge kernel

### PR DESCRIPTION
**Pull request by mwhudson**
This needs a lot more commentary and tests but its the end of my week and I think the logic is basically correct and I welcome any comments you have before I get back to it on Monday.

### CODE REVIEW

**Review by dbungert — COMMENTED** [review]
Looks reasonable so far

File: examples/sources/bridge.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        experience for operating your server.
+    id: ubuntu-server
+    locale_support: locale-only
+    name:
+      en: Ubuntu Server
+    path: ubuntu-server-minimal.ubuntu-server.squashfs
+    size: 1066115072
+    type: fsimage-layered
+    variant: server
+bridge_kernel_name: linux-generic-brg-22.04
```
[the remark this answers is not in the document]
Comment (mwhudson) [reply examples/sources/bridge.yaml:26]: oh yeah. the spec is also a bit inconsistent with latest terminology. I'll try to move the spec and this branch towards a consistent approach.

File: subiquity/models/source.py [hunk 2]
```diff
@@ -56,6 +57,19 @@ def __attrs_post_init__(self):
...
+class BridgeKernelReason(enum.Enum):
+    NVIDIA = "nvidia"
+    ZFS = "zfs"
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class SourceCatalog:
+    version: int
+    sources: typing.List[CatalogEntry]
+    bridge_kernel_name: typing.Optional[str] = None
```
[the remark this answers is not in the document]
Comment (mwhudson) [reply subiquity/models/source.py:69]: Yes I wondered about that but we already have a way to specify a default kernel, /etc/subiquity/kernel-meta-package (and similar). Should we work towards dropping that in favour of recording the default in the catalog? How would this interact with the server .2+ ISOs?

[the remark this answers is not in the document]
Comment (mwhudson) [reply subiquity/models/source.py:69]: Oh i see the spec covers this, good reading skills there Michael.

**Comment by mwhudson** [comment]
OK pushed a new version that has some tests and comments and is probably a bit better in most ways. Two things though:

1. I haven't thought about how this is going to interact with a kernel indicated by autoinstall or oem stuff (the latter probably not being a real concern in practice but I should think about it)
2. I think it would be better factored to replace the DRIVERS_LISTED event I add here with a DRIVERS_DECIDED event and move all the slightly painful logic around what happens if you say you want drivers before you know if there are any (which is what can happen on desktop) into the driver controller rather than having the kernel controller deal with it.

I'll try to work on those tomorrow.

**Comment by mwhudson** [comment]
I think this is conceptually sound now, and possibly ready for merging? It could stand to gain some more tests I think.

**Review by dbungert — APPROVED** [review]

**Pull request merged**